src/CodeGenerators/SettingsGen/Models/ParameterModel.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Serialization;
+
+namespace Microsoft.Omex.CodeGenerators.SettingsGen.Models
+{
+	/// <summary>
+	/// Parameter element used for service fabric settings
+	/// </summary>
+	public sealed class ParameterModel : IEqualityComparer<ParameterModel>, IEquatable<ParameterModel>
+	{
+		/// <summary>
+		/// Parameterless constructor for the Xml serialising
+		/// </summary>
+		public ParameterModel()
+		{
+		}
+
+		/// <summary>
+		/// Name attribute of the Parameter element
+		/// </summary>
+		[XmlAttribute]
+		public string Name { get; set; } = string.Empty;
+
+		/// <summary>
+		/// Value attribute of the Parameter element
+		/// </summary>
+		[XmlAttribute]
+		public string Value { get; set; } = string.Empty;
+
+		/// <summary>
+		/// MustOverride attribute of the Parameter element
+		/// </summary>
+		[XmlAttribute]
+		public string? MustOverride { get; set; }
+
+		/// <summary>
+		/// IsEncrypted attribute of the Parmater element
+		/// </summary>
+		[XmlAttribute]
+		public string? IsEncrypted { get; set; }
```

Comment:
What value could we have here, judging from the name I would assume it should be `bool`

Reply:
For the attribute properties they can't be boolean

Reply:
can't be nullable boolean

Reply:
And if it has a default value it will always output it

Reply:
Then what about using bool and skip adding attribute if it `false`. Or use different type for serialization and that would also help with Type property

Reply:
Also, this type not an attribute so you can have `bool?` property

Reply:
Xml fails to reflect for bool? also gonna keep it as string? because I'm having issues getting it to ignore default value